src/m365/entra/commands/pim/pim-role-assignment-eligibility-list.ts

```diff
@@ -0,0 +1,158 @@
+import { UnifiedRoleEligibilityScheduleInstance } from '@microsoft/microsoft-graph-types';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import { Logger } from '../../../../cli/Logger.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+import { validation } from '../../../../utils/validation.js';
+import { entraUser } from '../../../../utils/entraUser.js';
+import { entraGroup } from '../../../../utils/entraGroup.js';
+import { odata } from '../../../../utils/odata.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userId?: string;
+  userName?: string;
+  groupId?: string;
+  groupName?: string;
+  includePrincipalDetails?: boolean;
+}
+
+class EntraPimRoleAssignmentEligibilityListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.PIM_ROLE_ASSIGNMENT_ELIGIBILITY_LIST;
+  }
+
+  public get description(): string {
+    return 'Retrieves a list of eligible roles a user or group can be assigned to';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupName: typeof args.options.groupName !== 'undefined',
+        includePrincipalDetails: !!args.options.includePrincipalDetails
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: "--userId [userId]"
+      },
+      {
+        option: "--userName [userName]"
+      },
+      {
+        option: "--groupId [groupId]"
+      },
+      {
+        option: "--groupName [groupName]"
+      },
+      {
+        option: "--includePrincipalDetails [includePrincipalDetails]"
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
```

Comment:
Let's also add a validation to make sure that the specified userName is a valid user principal name.